File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationCancelEvent.java
package com.mapbox.services.android.navigation.v5.navigation.metrics;

import android.annotation.SuppressLint;

@SuppressLint("ParcelCreator")
class NavigationCancelEvent extends NavigationEvent {
  private static final String NAVIGATION_CANCEL = "navigation.cancel";
  private String arrivalTimestamp;
  private int rating;
  private String comment;

  NavigationCancelEvent(PhoneState phoneState) {
    super(phoneState);
  }

  @Override
  String getEventName() {
    return NAVIGATION_CANCEL;
  }

  String getArrivalTimestamp() {
    return arrivalTimestamp;
  }

  void setArrivalTimestamp(String arrivalTimestamp) {
    this.arrivalTimestamp = arrivalTimestamp;
  }

  int getRating() {
    return rating;
  }

  void setRating(Integer rating) {
    this.rating = rating;
  }

  String getComment() {
    return comment;
  }

  void setComment(String comment) {
    this.comment = comment;
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationCancelEvent.kt
package com.mapbox.services.android.navigation.v5.navigation.metrics

import android.annotation.SuppressLint

@SuppressLint("ParcelCreator")
internal class NavigationCancelEvent(phoneState: PhoneState) : NavigationEvent(phoneState) {
    var arrivalTimestamp: String? = null
    var rating: Int = 0
        private set
    var comment: String? = null

    override val eventName: String
        get() = NAVIGATION_CANCEL

    fun setRating(newRating: Int?) {
        newRating?.let {newRating->
            rating = newRating
        }
    }

    companion object {
        private val NAVIGATION_CANCEL = "navigation.cancel"
    }
}
